operatoringress/v1/types.go

```diff
@@ -0,0 +1,118 @@
+package v1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// DNSRecord is a DNS record managed in the zones defined by
+// dns.config.openshift.io/cluster .spec.publicZone and .spec.privateZone.
+//
+// Cluster admin manipulation of this resource is not supported. This resource
+// is only for internal communication of OpenShift operators.
+type DNSRecord struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the dnsecord.
+	Spec DNSRecordSpec `json:"spec"`
+	// status is the most recently observed status of the dnsRecord.
+	Status DNSRecordStatus `json:"status"`
+}
+
+// DNSRecordSpec contains the details of a DNS record.
+type DNSRecordSpec struct {
+	// dnsName is the hostname of the DNS record
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	DNSName string `json:"dnsName"`
+	// targets are record targets.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +required
+	Targets []string `json:"targets"`
+	// recordType is the DNS record type. For example, "A" or "CNAME".
+	// +kubebuilder:validation:Required
+	// +required
+	RecordType DNSRecordType `json:"recordType"`
+	// recordTTL is the record TTL in seconds. If <= zero, the default is 30.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	RecordTTL int64 `json:"recordTTL"`
+}
+
+// DNSRecordStatus is the most recently observed status of each record.
+type DNSRecordStatus struct {
+	// zones are the status of the record in each zone.
+	//
+	// +kubebuilder:validation:Optional
```

Comment:
the default is optional. You only need a marker on required items.

Reply:
Fixed here and one other place